doc/getting_started/installation_linux.rst

```diff
@@ -290,6 +290,9 @@ installed it.
    this allow Zephyr to pick the right toolchain, while allowing multiple Zephyr
    SDKs to be grouped together at a custom location.
 
+#. Set the required :ref:`environment variable <env_vars>`,
+   :envvar:`ZEPHYR_TOOLCHAIN_VARIANT` to ``zephyr``.
```

Comment:
Are you sure this is really required? https://github.com/zephyrproject-rtos/zephyr/blob/b49292634fc938855a535c89d7f3cdb432e07a7f/cmake/toolchain/zephyr/host-tools.cmake#L30

Reply:
If you don't `source zephyr-env.sh` it is. Or at least it was when the bug was filed! @tejlmand

Reply:
If it's not required anymore then we do not need this PR at all, because then #30713 is moot?

Reply:
Hmm, hold on. Could this be a `twister` issue?

`scripts/tests/twister/test_testsuite_class.py:    assert out == "E: Variable ZEPHYR_TOOLCHAIN_VARIANT is not defined\n"`

@nashif apparently `ZEPHYR_TOOLCHAIN_VARIANT` is not required in general, but it is for twister, which is what Andrew ran in #30713

Reply:
> @nashif apparently `ZEPHYR_TOOLCHAIN_VARIANT` is not required in general, but it is for twister, which is what Andrew ran in #30713

how is that possible that we do not need it? I keep switching between toolchains and I am sure I need it unless I am missing something fundamental here.

Reply:
Didn't some @tejlmand magic make this unnecessary?

This works for me:

```
$ unset ZEPHYR_GCC_VARIANT; unset ZEPHYR_TOOLCHAIN_VARIANT; unset ZEPHYR_SDK_INSTALL_DIR; unset TOOLCHAIN_ROOT; unset CROSS_COMPILE; cmake -B build -S samples/hello_world -GNinja -DBOARD=nrf52840dk_nrf52840
Including boilerplate (Zephyr repository): /home/mbolivar/zp/zephyr/cmake/app/boilerplate.cmake
-- Application: /home/mbolivar/zp/zephyr/samples/hello_world
-- Zephyr version: 2.5.0-rc2 (/home/mbolivar/zp/zephyr)
-- Found Python3: /home/mbolivar/.virtualenvs/west-0.9/bin/python3.9 (found suitable exact version "3.9.1") found components: Interpreter 
-- Found west (found suitable version "0.9.0", minimum required is "0.7.1")
-- Board: nrf52840dk_nrf52840
-- Cache files will be written to: /home/mbolivar/.cache/zephyr
ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found toolchain: zephyr (/home/mbolivar/bin/zephyr-sdk-0.11.3)
```

Reply:
It basically defaults to `zephyr` if not set is what I understand.

Reply:
Correct, it will default to `zephyr` if not set.

And it will look for the Zephyr SDK in those locations:
https://github.com/zephyrproject-rtos/zephyr/blob/4d4a636555cdec8785652e016d974667be7c201b/cmake/toolchain/zephyr/host-tools.cmake#L53-L60

When user installs the SDK, and decides to install outside of those search paths, he will be prompted with:
https://github.com/zephyrproject-rtos/sdk-ng/blob/7ef15bd022c46bd92f12576a5c51dca913507949/scripts/template_dir#L175
```
Do you want to register Zephyr-sdk CMake module with path <Zephyr-SDK install path> (y/n)? 
```

Reply:
> how is that possible that we do not need it? I keep switching between toolchains and I am sure I need it unless I am missing something fundamental here.

Apparently you always need it when using twister.
You don't need it when just building a sample using `west` (CMake / ninja) and the `zephyr` toolchain.
When switching between toolchains, like `gnuarmemb`, `xcc`, `xtools`, `arcmwdt` you would need to set it for those, and then you might as well set it for the `zephyr` toolchain as well.

https://github.com/zephyrproject-rtos/zephyr/blob/4d4a636555cdec8785652e016d974667be7c201b/scripts/pylib/twister/twisterlib.py#L2811-L2826

Will try to see if we can align how the toolchain is decided, so we don't need to keep twister and CMake in sync.